kratos/modified_shape_functions/tetrahedra_3d_4_ausas_incised_shape_functions.h

```diff
@@ -0,0 +1,215 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+#if !defined(KRATOS_TETRAHEDRA_3D_4_AUSAS_INCISED_SHAPE_FUNCTIONS)
+#define KRATOS_TETRAHEDRA_3D_4_AUSAS_INCISED_SHAPE_FUNCTIONS
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "modified_shape_functions/tetrahedra_3d_4_ausas_modified_shape_functions.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+class KRATOS_API(KRATOS_CORE) Tetrahedra3D4AusasIncisedShapeFunctions : public Tetrahedra3D4AusasModifiedShapeFunctions
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of Tetrahedra3D4AusasModifiedShapeFunctions
+    KRATOS_CLASS_POINTER_DEFINITION(Tetrahedra3D4AusasIncisedShapeFunctions);
+
+    // General type definitions
+    typedef AusasModifiedShapeFunctions                        BaseType;
+    typedef BaseType::GeometryPointerType                      GeometryPointerType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor
+    Tetrahedra3D4AusasIncisedShapeFunctions(const GeometryPointerType rpInputGeometry,
+        const Vector& rNodalDistancesWithExtrapolated, const Vector& rExtrapolatedEdgeRatios);
+
+    /// Destructor
+    ~Tetrahedra3D4AusasIncisedShapeFunctions();
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override;
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override;
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override;
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+    * Returns a reference to the extrapolated edge ratios vector member variable.
+    */
+    const Vector& GetExtrapolatedEdgeRatios() const ;
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    // Arrays to get edge and node IDs of geometry from edge ID of splitting utility
+    const size_t edge_id_for_geometry [6] = {0, 2, 3, 1, 4, 5};
+    const size_t node_ids_for_geometry [6][2] = {{0,1}, {2,0}, {0,3}, {1,2}, {1,3}, {2,3}};
```

Comment:
We tend to avoid using c++ arrays. Couldn't we use an std::array to achieve the same?